internal/provider/resource_aws_cloud_provider_credential.go

```diff
@@ -0,0 +1,148 @@
+package provider
+
+import (
+	"context"
+
+	"github.com/PacketFabric/terraform-provider-packetfabric/internal/packetfabric"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceCloudProviderCredentialAws() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceCloudProviderCredentialAwsCreate,
+		ReadContext:   resourceCloudProviderCredentialRead,
+		UpdateContext: resourceCloudProviderCredentialAwsUpdate,
+		DeleteContext: resourceCloudProviderCredentialDelete,
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"description": {
+				Type:         schema.TypeString,
+				Required:     true,
```

Comment:
description is not required in the API

Reply:
I will keep it required in TF, otherwise, it shows empty in the UI.